onnxruntime/core/framework/session_options.h

```diff
@@ -181,6 +181,17 @@ struct SessionOptions {
   void AddCustomOpLibraryHandle(PathString library_name, void* library_handle);
 #endif
 
+  // Load cancellation flag is necessary to be within shared memory as session_options are
+  std::shared_ptr<bool> load_cancellation_flag = std::make_shared<bool>(false);
```

Comment:
I don't understand why it needs to be `shared_ptr<bool>` within the SessionOptions struct.
also, given we are potentially accessing this flag from multiple threads, should we use `std::atomic<bool>`?

Reply:
This needs to be shared, because inference session creates a copy the options. There should be a shared piece of memory between the session copy and the user.

I do not feel this needs to be so precise and we do not need the atomic. The RunOptions works in the same way. I can make it atomic though.